src/main/scala/io/iohk/ethereum/blockchain/sync/regular/BlockImporter.scala

```diff
@@ -389,6 +400,26 @@ object BlockImporter {
   case class ResolvingMissingNode(blocksToRetry: NonEmptyList[Block]) extends NewBehavior
   case class ResolvingBranch(from: BigInt) extends NewBehavior
 
+  sealed trait BlockImportType {
+    def recordMetric(time: Long): Unit
+  }
+
+  case object MinedBlockImport extends BlockImportType {
+    override def recordMetric(time: Long): Unit = RegularSyncMetrics.recordMinedBlockPropagationTimer(time)
+  }
+
+  case object CheckpointBlockImport extends BlockImportType {
+    override def recordMetric(time: Long): Unit = RegularSyncMetrics.recordImportCheckpointPropagationTimer(time)
+  }
+
+  case object NewBlockImport extends BlockImportType {
+    override def recordMetric(time: Long): Unit = RegularSyncMetrics.recordImportNewBlockPropagationTimer(time)
+  }
+
+  case object DefaultBlockImport extends BlockImportType {
+    override def recordMetric(time: Long): Unit = RegularSyncMetrics.recordDefaultBlockPropagationTimer(time)
```

Comment:
`nanos` would also be better here (and two more just above)

Reply:
Fixed!